src/Entities/Player.cpp

```diff
@@ -220,8 +222,29 @@ void cPlayer::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 	m_Stats.AddValue(statMinutesPlayed, 1);
 
+	if (m_IsFrozen)
+	{
+		m_FreezeCounter += 1;
+		if (!m_IsManuallyFrozen && a_Chunk.IsValid())
+		{
+			Unfreeze();
+		}
+		else
+		{
+			SetPosition(m_FrozenPosition);
+			SetSpeed(0, 0, 0);
+			if (m_FreezeCounter % 60 == 0)
+			{
+				BroadcastMovementUpdate(m_ClientHandle.get());
+				m_ClientHandle->SendPlayerPosition();
+			}
+			return;
+		}
+	}
+
```

Comment:
This new block of code could use some comment - why it's there, what's it doing (high-level)